sklearn/metrics/_classification.py

```diff
@@ -2272,7 +2326,7 @@ def classification_report(
 
         .. versionadded:: 0.20
 
-    zero_division : "warn", 0 or 1, default="warn"
+    zero_division : {"warn", 0.0, 1.0, np.nan}, default="warn"
         Sets the value to return when there is a zero division. If set to
         "warn", this acts as 0, but warnings are also raised.
```

Comment:
We should add the info about `np.nan` in the description and add the `versionadded` directive.

```suggestion
        "warn", this acts as 0, but warnings are also raised.

        .. versionadded:: 1.3
           `np.nan` option was added.
```